src/shapes/text.class.js

```diff
@@ -741,23 +757,49 @@
     _measureLine: function(lineIndex) {
       var width = 0, i, grapheme, line = this._textLines[lineIndex], prevGrapheme,
           graphemeInfo, numOfSpaces = 0, lineBounds = new Array(line.length),
-          positionInPath = 0, startingPoint, totalPathLength, path = this.path;
+          positionInPath = 0, startingPoint, totalPathLength, path = this.path,
+          reverse = this.side == "right" || this.direction == "rtl";
 
       this.__charBounds[lineIndex] = lineBounds;
       if (path) {
         startingPoint = fabric.util.getPointOnPath(path.path, 0, path.segmentsInfo);
         totalPathLength = path.segmentsInfo[path.segmentsInfo.length - 1].length;
         startingPoint.x += path.pathOffset.x;
         startingPoint.y += path.pathOffset.y;
+        for (i = 0; i < line.length; i++) {
+          grapheme = line[i];
+          graphemeInfo = this._getGraphemeBox(grapheme, lineIndex, i, prevGrapheme);
+          lineBounds[i] = graphemeInfo;
+          width += graphemeInfo.kernedWidth;
+          prevGrapheme = grapheme;
+        }
+        switch(this.textAlign) {
+          case "center":
+            positionInPath = (totalPathLength - width) / 2;
+          break;
+          case "right":
+            positionInPath = (totalPathLength - width);
+          break;
+          //justify not yet supported
+        }
+        if (this.direction == "ltr") {
+          positionInPath += this.startOffset;
+        }
+        else if (this.direction == "rtl") {
+          positionInPath -= this.startOffset;
+        }
       }
-      for (i = 0; i < line.length; i++) {
+      for (i = reverse ? line.length - 1 : 0; reverse ? i >= 0 : i < line.length; reverse ? i-- : i++) {
```

Comment:
why you want to loop through graphemes in reverse?

Reply:
It's reversed when the `side` property is 'right` or `direction` is `rtl`

Reply:
This seemed like the simpler approach than figuring out how to reverse the path itself. Would you approach this differently?